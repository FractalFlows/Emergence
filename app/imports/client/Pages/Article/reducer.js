/*
 * Built by Astrocoders
 * @flow
 */

import {
  LOAD_RELATED_ARTICLES_RESULT,
  SEARCH_RELATED_ARTICLES,
  CLEAR_RELATEDS_SEARCH,
} from './constants'

const initialState = {
  relatedArticles: [],
  isSearchingForRelatedArticles: false,
}

export default function article(state = initialState, { type, payload }) {
  switch (type) {
    case SEARCH_RELATED_ARTICLES:
      return {
        ...state,
        isSearchingForRelatedArticles: true,
      }
    case LOAD_RELATED_ARTICLES_RESULT:
      return {
        ...state,
        isSearchingForRelatedArticles: false,
        relatedArticles: payload.result,
      }
    case CLEAR_RELATEDS_SEARCH:
      return {
        ...state,
        isSearchingForRelatedArticles: false,
        relatedArticles: [],
      }

    default:
      return { ...state }
  }
}
