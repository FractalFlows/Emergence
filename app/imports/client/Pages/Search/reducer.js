import {
  FETCH_SEARCH,
  LOAD_SEARCH_RESULTS,
  CLEAR_SEARCH,
  FETCH_SELECTED_ARTICLE,
  LOAD_SELECTED_ARTICLE,
} from './constants'

type Article = {
  slug: string,
  DOI: string,
  abstract: string,
  authors: Array<string>,
}

type State = {
  articlesFromSearch: Array<Article>,
  searchText: string,
  isSearch: bool,
}

const initialState = {
  articlesFromSearch: [],
  searchText: '',
  isSearching: false,
}

export default function search(state: State = initialState, { type, payload }: Action){
  switch(type){
    case FETCH_SEARCH: 
      return {
        ...state,
        isSearching: true,
        searchText: payload.searchText,
      }
    case LOAD_SEARCH_RESULTS:
      return {
        ...state,
        isSearching: false,
        articlesFromSearch: payload.articles,
      }
    case CLEAR_SEARCH: 
      return {
        ...state,
        searchText: '',
        articlesFromSearch: [],
        isSearching: false,
      }
    case FETCH_SELECTED_ARTICLE: 
      return {
        ...state,
        isFetchingSelectedArticle: true,
      }
    case FETCH_SELECTED_ARTICLE: 
      return {
        ...state,
        isFetchingSelectedArticle: true,
      }
    case LOAD_SELECTED_ARTICLE: 
      return {
        ...state,
        isFetchingSelectedArticle: false,
      }
    default: return { ...state }
  }
} 
