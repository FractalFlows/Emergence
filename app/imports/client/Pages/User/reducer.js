/*
 * Built by Astrocoders
 * @flow
 */

const initialState = {

}

type State = {
}

type Action = {
  type: string,
  payload: any,
}

export default function user(state:State = initialState, { type, payload }:Action):State {
  switch (type) {
    default:
      return state
  }
}
