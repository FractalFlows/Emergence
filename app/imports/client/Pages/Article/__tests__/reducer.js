import reducer from '../reducer'
import {
  searchRelatedArticles,
  loadRelatedsSearchResult,
  clearRelatedsSearch,
} from '../actions'

test('should handle default', () => {
  const state = reducer({}, {})

  expect(state).toEqual({})
})

test('should handle SEARCH_RELATED_ARTICLES action', () => {
  const action = searchRelatedArticles({
    articleSlug: 'article-some',
  })
  const state = reducer({}, action)

  expect(state.isSearchingForRelatedArticles).toBeTruthy()
  expect(state).toMatchSnapshot()
})

test('should handle LOAD_RELATED_ARTICLES_RESULT action', () => {
  const action = loadRelatedsSearchResult({ result: [] })
  const state = reducer({}, action)

  expect(state.isSearchingForRelatedArticles).toBeFalsy()
  expect(state.relatedArticles).toEqual([])
  expect(state).toMatchSnapshot()
})

test('should handle CLEAR_RELATEDS_SEARCH action', () => {
  const action = clearRelatedsSearch()
  const state = reducer({ relateds: [{slug: 'something'}] }, action)

  expect(state.relatedArticles).toEqual([])
  expect(state).toMatchSnapshot()
})
