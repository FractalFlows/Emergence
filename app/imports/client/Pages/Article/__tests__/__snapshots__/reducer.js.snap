exports[`test should handle CLEAR_RELATEDS_SEARCH action 1`] = `
Object {
  "isSearchingForRelatedArticles": false,
  "relatedArticles": Array [],
}
`;

exports[`test should handle LOAD_RELATED_ARTICLES_RESULT action 1`] = `
Object {
  "isSearchingForRelatedArticles": false,
  "relatedArticles": Array [],
}
`;

exports[`test should handle SEARCH_RELATED_ARTICLES action 1`] = `
Object {
  "isSearchingForRelatedArticles": true,
}
`;
